### PR TITLE
Define shortcut types for typed list

### PR DESCRIPTION
**Pull request by Asalle**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [X] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [X] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Nosferatul** [comment]
@Asalle please run `coala` locally in your repo directory.

File: coalib/settings/Setting.py [hunk 1]
```diff
@@ -65,6 +65,13 @@ def __repr__(self):
 
     return Converter()
 
+def str_list():
+    """typed_list shortcut for string"""
+    return typed_list(str)
+
+def int_list():
+    """typed_list shortcut for int"""
```
Comment (Asalle) [inline coalib/settings/Setting.py]: Accepted

File: coalib/settings/Setting.py [hunk 2]
```diff
@@ -65,6 +65,13 @@ def __repr__(self):
 
     return Converter()
 
+def str_list():
+    """typed_list shortcut for string"""
```
Comment (Asalle) [inline coalib/settings/Setting.py]: Accepted

File: coalib/settings/Setting.py [hunk 3]
```diff
@@ -65,6 +65,13 @@ def __repr__(self):
...
+    return typed_list(str)
+
+def int_list():
+    """typed_list shortcut for int"""
+    return typed_list(int)
 
 def typed_dict(key_type, value_type, default):
```
Comment (Asalle) [inline coalib/settings/Setting.py]: Accepted

File: coalib/settings/Setting.py [hunk 4]
```diff
@@ -65,6 +65,13 @@ def __repr__(self):
 
     return Converter()
 
+def str_list():
+    """typed_list shortcut for string"""
+    return typed_list(str)
+
+def int_list():
```
Comment (Asalle) [inline coalib/settings/Setting.py]: Accepted

File: coalib/settings/Setting.py [hunk 5]
```diff
@@ -65,6 +65,13 @@ def __repr__(self):
 
     return Converter()
 
+def str_list():
```
Comment (Asalle) [inline coalib/settings/Setting.py]: Accepted

File: coalib/settings/Setting.py [hunk 6]
```diff
@@ -66,6 +66,16 @@ def __repr__(self):
     return Converter()
 
 
+def str_list():
+    '""typed_list shortcut for string""'
+    return typed_list(str)
```
Comment (Makman2) [inline coalib/settings/Setting.py]: I would rather do this:
```
str_list = typed_list(str)
```
Then the user doesn't have to use redundant braces like: `str_list()`

**Comment by Makman2** [comment]
I've deleted the gitmate comments as you've addressed them :+1:

**Comment by Makman2** [comment]
Sorry accidentally closed :3

**Comment by Makman2** [comment]
Also tests are missing :+1:

**Comment by Asalle** [comment]
Added tests :)
Also, should I add shortcuts for all the datatypes specified here https://github.com/coala/coala/blob/master/docs/Developers/Writing_Native_Bears.rst#what-data-types-are-supported?

**Comment by Makman2** [comment]
yeah sounds good :)

**Comment by Makman2** [comment]
About commit message:
1. Please write `Closes <full-link-to-issue>` instead of just issue number
2. You can leave the part "and a ut" in the shortlog. Adding tests is common, so you don't have to write that explicitly :)

**Comment by Asalle** [comment]
Can I do something to resolve conflicts?

**Comment by Makman2** [comment]
You have to resolve them manually after you've initiated a rebase. Google a bit around to see how manually resolution of conflicts work in git :+1:

File: docs/Developers/Writing_Native_Bears.rst [hunk 7]
```diff
@@ -261,6 +261,8 @@ The Setting does support some very basic types:
 -  List of strings (``list``, values will be split by comma)
 -  Dict of strings (``dict``, values will be split by comma and colon)
 
+You can use shortcuts for basic types, for example, ``str_list`` for String.
```
Comment (Makman2) [inline docs/Developers/Writing_Native_Bears.rst]: Rather list them all :+1:

File: tests/settings/SettingTest.py [hunk 8]
```diff
@@ -100,6 +100,43 @@ def test_typed_list(self):
...
+
+        # bool
+        self.uut = Setting('key', 'true, nope, yeah')
+        self.assertEqual(bool_list(self.uut), [True, False, True])
+
+        with self.assertRaises(ValueError):
+            self.uut = Setting('key', 'true, false, 78, 89.0')
+            bool_list(self.uut)
+
+        self.assertEqual(repr(bool_list), 'typed_list(bool)')
```
Comment (Makman2) [inline tests/settings/SettingTest.py]: you should move each subtests for your specialized types into an own test function. E.g. `test_str_list`, `test_int_list` etc.
Using separate test functions makes test design more modular :+1:

File: tests/settings/SettingTest.py [hunk 9]
```diff
@@ -8,7 +8,7 @@
     Setting, path, path_list, url, typed_dict, typed_list, typed_ordered_dict,
     glob, glob_list,
     language,
-)
+    float_list, bool_list, int_list, str_list)
```
Comment (Makman2) [inline tests/settings/SettingTest.py]: maybe you should keep the old code-style for imports, so leaving the closing `)` on a new line and having a trailing comma:
```
    language,
    float_list, bool_list, int_list, str_list,
)
```

**Comment by Makman2** [comment]
Commit message: I would rephrase shortlog to
```
Setting.py: Add shortcuts for typed_list

For str, int, float and bool types.

Closes XXXX
```
I think the information about `typed_list` is essential :)

**Comment by Asalle** [comment]
Is it a random travis ci build fail? I don't see how it's connected to my changes.

**Comment by adhikasp** [comment]
Possibly, I restarted the job.

File: tests/settings/SettingTest.py [hunk 10]
```diff
@@ -100,6 +101,41 @@ def test_typed_list(self):
...
+        self.uut = Setting('key', '0.8, 1.3, 5.87')
+        self.assertEqual(float_list(self.uut), [0.8, 1.3, 5.87])
+
+        with self.assertRaises(ValueError):
+            self.uut = Setting('key', '1.987, a, 3')
+            float_list(self.uut)
+
+        self.assertEqual(repr(float_list), 'typed_list(float)')
+
+    def test_float_list(self):
```
Comment (Makman2) [inline tests/settings/SettingTest.py]: you probably meant `test_bool_list` ;)

File: docs/Developers/Writing_Native_Bears.rst [hunk 11]
```diff
@@ -261,6 +261,9 @@ The Setting does support some very basic types:
 -  List of strings (``list``, values will be split by comma)
 -  Dict of strings (``dict``, values will be split by comma and colon)
 
+You can use shortcuts for basic types, ``str_list`` for String,
+``int_list`` for Int, ``float_list`` for Float and ``bool_list`` for Boolean.
```
Comment (Makman2) [inline docs/Developers/Writing_Native_Bears.rst]: also refer to the python types using backticks, so:
```
You can use shortcuts for basic types, ``str_list`` for ``str`` ,
``int_list`` for ``int``, ``float_list`` for ``float`` and ``bool_list`` for ``bool``.
```
Or write them out if you don't want to refer to the python types:

```
You can use shortcuts for basic types, ``str_list`` for strings,
``int_list`` for integers, ``float_list`` for floats and ``bool_list`` for boolean values.
```

**Comment by Asalle** [comment]
@Makman2 review BADLY needed xD

**Comment by Makman2** [comment]
ack 38f5b78

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@rultor merge
